db/version_set.h

```diff
@@ -695,12 +695,22 @@ class VersionSet {
     return last_sequence_.load(std::memory_order_acquire);
   }
 
+  uint64_t LastToBeWrittenSequence() const {
+    return last_to_be_written_sequence_.load(std::memory_order_acquire);
+  }
+
   // Set the last sequence number to s.
   void SetLastSequence(uint64_t s) {
     assert(s >= last_sequence_);
+    assert(s <= last_to_be_written_sequence_);
     last_sequence_.store(s, std::memory_order_release);
   }
 
+  void SetLastToBeWrittenSequence(uint64_t s) {
+    assert(s >= last_to_be_written_sequence_);
+    last_to_be_written_sequence_.store(s, std::memory_order_release);
```

Comment:
SetLastToBeWrittenSequence is used only during initialization so its stronger ordering guarantee does not make much difference in performance. The newly added FetchAddLastToBeWrittenSequence could be accessed outside mutex and its ordering guarantees cannot be relaxed.